doc/usecases/record_elektra/UC_ansible-libelektra_marked_keys.md

```diff
@@ -0,0 +1,18 @@
+# Use Case: Fail application of ansible-libelektra if certain keys differ in value
+
+## Summary
+
+- **Title:** Fail application of ansible-libelektra if certain keys differ in value
+- **Scope:** ansible-libelektra
+- **Level:** User goal
+- **Actors:** User (usually sysadmin)
+- **Brief:** We should be able to mark certain keys in the tasks for a specific value. If the keys do have a different value on the target machine, ansible-libelektra should fail.
```

Comment:
It is still not clear to me why we the system administrator ansible-libelektra to fail? What is the goal? Give a bit background why the user needs this. Give examples.

Reply:
This can be used as a safe guard to ensure the configuration is only applied to the correct machines. Probably won't be used by many, but it could be useful to some. This the feature needed for `kdb record mark` to make any sense.

Reply:
A UC should clearly state for whom, in which situation and why it is useful.

Obviously we cannot know how often features will actually be used. For now there should be at least one convincing example where we ourselves would use it (or at least can imagine using it).

Reply:
I think one example would be a dependency between two applications. Let's say apps A and B work together. There could a key in A's config that may be set to tell A to work with B. Now we define a config for B and this config only works, when used with A. Therefore, we need A to configured to work with B.

We could of course just set A's key to the right value and include that in the recording. But IMO that would really be good practice. We'd be reconfiguring A when we apply the config for B. Instead a better solution would be just marking the the key in A's config and thereby recording that it must be set correctly for this recording to be applicable.

---

Another example that isn't covered by this UC, but which might be useful (and is probably more common): Let's say there is a key that can't be set automatically, but must be manually configured by the sysadmin before applying the config. This could for example be a path that is unique to the machine. In this case, we'd need away to record "this key must be set, but we don't care about the value".

However, this special case can probably be better achieved, by adding an additional task to the Ansible playbook.

---

The above are the most reasonable examples I could think of, but there are other more questionable cases. For example, if setting the key in question is not idempotent, e.g. because of some (less than ideal) plugin used for the mountpoint.

---

> **Note** On a technical level, the ability to record such "marked keys" falls directly out of the "record old & new values" format. For a marked key, the old and new value would just be identical. Therefore, what we are really deciding on with this use case is whether or not to expose this feature to the user.

Reply:
Yes, I agree: limited/broken plugins, in particular read-only plugins, are a situation I could imagine it makes sense. There are some cases where plugins must be read-only by design (hardware-information, read-only file systems etc.) so in these cases assertions are the only way.

Btw. terminology-wise we used "assert" in #4399. This alone would make the UC much clearer.

Reply:
Another example would be to check if a certain Elektra plugin is installed, e.g. the `check` plugin.